adapters/repos/db/lsmkv/compactor_replace.go

```diff
@@ -81,10 +85,15 @@ func (c *compactorReplace) do() error {
 		dataEnd = uint64(kis[len(kis)-1].ValueEnd)
 	}
 
-	if err := c.writeHeader(c.currentLevel, 0, c.secondaryIndexCount, dataEnd); err != nil {
+	if err := c.writeHeader(segmentFile, c.currentLevel,
+		segmentindex.SegmentV1, c.secondaryIndexCount, dataEnd); err != nil {
```

Comment:
good, so even if compaction takes a (v0,v0) or (v0,v1) or (v1,v0) segment pair it will end up generating a v1 segment